src/ansi-c/goto-conversion/goto_convert_class.h

```diff
@@ -56,6 +56,25 @@ class goto_convertt : public messaget
   std::string tmp_symbol_prefix;
   lifetimet lifetime = lifetimet::STATIC_GLOBAL;
 
+  struct clean_expr_resultt
+  {
+    std::list<irep_idt> temporaries;
+    goto_programt side_effects;
```

Comment:
is `side_effects` the program resulting from cleaning the side effect expression ? How is the expression that represents the `side_effect_exprt` itself returned ?

Reply:
The expression itself is still modified in place, I have not (yet) changed this behaviour of `clean_expr`. Likely this should also be changed, but I'd prefer to do so in a follow-up PR as the changes will look a bit different and will no longer necessarily touch the same lines. (We often clean some operand of an expression and would then want to only conditionally update the original expression when there actually was a change. Somewhat similar to how we do it in the simplifiers nowadays.)